### ISSUE TITLE
Simple Trame App Example not Working

### ISSUE THREAD

**Issue by Rmelikyan** [pyvista/pyvista#5430]
### Describe the bug, what's wrong, and what you expected.

This may be a documentation issue or user error as a total Vue/Vuetify beginner, but the [pyvista trame example](https://tutorial.pyvista.org/tutorial/09_trame/a_trame_simple.html) no longer works for me since updating Vue on my system such that the default is vue3 instead of vue2. I've tried modifying the example to get the server to run, but then I run into a new issue where there is a JS Error. Screenshots and code below.

### Steps to reproduce the bug.

a_trame_simple.py (as currently provided in the pyVista tutorial)
```python
import pyvista as pv
from pyvista import examples
from pyvista.trame.ui import plotter_ui
from trame.app import get_server
from trame.ui.vuetify import SinglePageLayout

pv.OFF_SCREEN = True

server = get_server()
state, ctrl = server.state, server.controller

mesh = examples.load_random_hills()

pl = pv.Plotter()
pl.add_mesh(mesh)

with SinglePageLayout(server) as layout:
    with layout.content:
        view = plotter_ui(pl)

server.start()
```
This results in the following error: 
> TypeError: Server using client_type='vue3' while we expect 'vue2'

previously (on pyvista v 0.42.3) this example ran as expected.

now, if I modify the above example to
`from trame.ui.vuetify import SinglePageLayout`
The server does start! but it does not look as expected.
Here is the resulting screen:
<img width="1595" alt="Screen Shot 2024-01-02 at 9 19 44 PM" src="https://github.com/pyvista/pyvista/assets/57904723/6028d40f-9233-4959-b6d1-aa8a6f7833cb">
And a snippet of the errorlog (where the JS Error prints continuously until killed):

> python snippets/a_trame_simple.py

> And to prevent your browser from opening, add '--server' to your command line.
> 
> App running at:
>  - Local:   http://localhost:8080/
>  - Network: http://127.0.0.1:8080/
> 
> Note that for multi-users you need to use and configure a launcher.
>  JS Error => Lookup error: Vue.use(Vuetify.createVuetify(), {})
>  JS Error => TypeError: Cannot set properties of undefined (setting 'view_P_0x125dd4310_0')
>  JS Error => error: Uncaught TypeError: Cannot read properties of null (reading '0')
>  JS Error => TypeError: Cannot set properties of undefined (setting 'view_P_0x125dd4310_0')
>  JS Error => TypeError: Cannot set properties of undefined (setting 'view_P_0x125dd4310_0')
>  JS Error => TypeError: Cannot set properties of undefined (setting 'view_P_0x125dd4310_0')
>  JS Error => TypeError: Cannot set properties of undefined (setting 'view_P_0x125dd4310_0')
>  JS Error => TypeError: Cannot set properties of undefined (setting 'view_P_0x125dd4310_0')

### System Information

```shell
--------------------------------------------------------------------------------
  Date: Tue Jan 02 21:26:12 2024 EST

                OS : Darwin
            CPU(s) : 16
           Machine : x86_64
      Architecture : 64bit
               RAM : 64.0 GiB
       Environment : Python
       File system : apfs
        GPU Vendor : ATI Technologies Inc.
      GPU Renderer : AMD Radeon Pro 5500M OpenGL Engine
       GPU Version : 4.1 ATI-4.6.21
  MathText Support : True

  Python 3.9.17 (main, Jun 20 2023, 17:03:39)  [Clang 13.0.0
  (clang-1300.0.29.30)]

           pyvista : 0.43.1
               vtk : 9.3.0
             numpy : 1.23.3
        matplotlib : 3.7.2
            scooby : 0.7.4
             pooch : v1.6.0
            pillow : 9.2.0
           IPython : 8.5.0
        ipywidgets : 8.1.1
             scipy : 1.9.1
              tqdm : 4.64.1
             trame : 3.5.0
      trame_client : 2.14.2
      trame_server : 2.14.0
         trame_vtk : 2.6.3
     trame_vuetify : 2.3.1
      nest_asyncio : 1.5.6
--------------------------------------------------------------------------------
```

**Comment by jourdain** [comment]
I guess it is a documentation issue due to the change of the default and purposely annoying warning for 6 months. ;-)

Just replace `server = get_server()` by `server = get_server(client_type="vue2")`.

**Comment by jourdain** [comment]
@annehaley can you fix the tutorial to ensure the actual __client_type__ is provided when needed?

**Comment by jourdain** [comment]
Unless those example are so trivial that you can migrate them to vuetify3?

**Comment by Rmelikyan** [comment]
Hi Jourdain, thanks for responding!

I'm sure that I'm just confused and there is an error in my system configuration, but just in case, I'll give you a clear report about what I have found to work and not work for me.

This morning I've been going through the [trame tutorial](https://kitware.github.io/trame/guide/tutorial/setup.html) both locally on my mac and remotely on a linux server running ubuntu.

For each machine I get pretty much the opposite problem. 

Locally, after todays reboot and (yesterdays mad scramble to download and setup vue-cli to try to make vue3 work), using `from trame.ui.vuetify3 import SinglePageLayout` with `server = get_server(client_type="vue3")` (or ignoring the client_type argument completely) results in a working as expected trame app on my browser. If I switch to `from trame.ui.vuetify import SinglePageLayout` with `server = get_server(client_type="vue2")` I get the following error
>  JS Error => error: Uncaught TypeError: r.a.extend is not a function

and  an endless loading trame screen.

Remotely, the opposite. `vuetify3` with `"vue3"` results in endless loading screen though no error message printed to terminal, and `vuetify` with `"vue2"`  (as instructed by the trame tutorial) works and I can tunnel the remote app to my local browser.

Also, the remote server is able to run the above "a_trame_simple.py" with the inclusion of `client_type = "vue2"` as per your suggestion and the inclusion (and download) of `pv.start_xvfb()`

Long story short, I don't know why both machines are experiencing separate issues, but I can get each of them to work with either vue2 or vue3 (not both). Ideally both machines would work with the same vue client so I could develop independently but at least something is working and I can start to focus on the visualizations I actually want to develop.

Please let me know if you have any more insights! I think I'm good to close, but I'll leave that to you in case you want to update the documentation.

**Comment by annehaley** [comment]
I had been experiencing the same error when I made a test application first with vue 2 and then switched to vue 3. @jourdain  pointed out to me that the problem was actually in the browser caching. The browser had cached the application code for vue 2 and does not recognize that it should discard that cached code when the application is switched to vue 3.

From your description of your two machine tests, it is likely that one cached vue 2-compliant code first and the other cached vue 3-compliant code. Since one is local and one is remote, they each probably have their own domain and their own unshared cache.

Try clearing your browser cache before loading your updated trame application.

**Comment by jourdain** [comment]
One way to prevent issue between vue2/vue3 setup is to not use the same endpoint (localhost:8080). 

For vue3 apps, you can provide a `--port 8083` and make sure you stay consistent. Otherwise you will run into browser caching issue.

**Comment by Rmelikyan** [comment]
Wow! It's working!

I'm really glad I asked about this. I never would have guessed that this was a caching issue. Now that I know this I can be consistent about my port choices throughout my apps. Thank you both so much!

**Comment by tkoyama010** [comment]
> I guess it is a documentation issue due to the change of the default and purposely annoying warning for 6 months. ;-)
> 
> Just replace `server = get_server()` by `server = get_server(client_type="vue2")`.

In `main`, we have confirmed that this issue has not been resolved. Should we make this fix as an interim measure?

**Comment by jourdain** [comment]
yes we should

**Comment by LTMeyer** [comment]
Hello everyone,

I'm facing two issues:
- I've followed the [example](https://tutorial.pyvista.org/tutorial/09_trame/index.html) on pyvista and trame. As it is, the example raises the following error:
> TypeError: Server using client_type='vue3' while we expect 'vue2'

If I update the server for 
```python
server = get_server(client_type="vue2")
```
The example works as expected.
![image](https://github.com/pyvista/pyvista/assets/13559010/31d34c6a-a0d8-4471-b8cb-84df0c42ca68)

- The other issue relates to vue3. If in the example above I keep using vu3 by applying the following change, instead of reverting to vue2:
 ```python
 from trame.ui.vuetify3 import SinglePageLayout
 ```
 The Pyvista action bar does not appear anymore.
 
![image](https://github.com/pyvista/pyvista/assets/13559010/746e02b1-0e60-41f3-b5c8-6371ecbb6b47)

How could the example edited? How could I make it work with the action bar using v3?

**Comment by jourdain** [comment]
A fix is already merged in master for the vue3 toolbar but no release integrate it yet. 
For the example, indeed we should update it. Thanks for pointing that out. PRs are welcome.
